src/resmom/linux/alps.c

```diff
@@ -725,12 +796,10 @@ stack_busted(ud_t *d)
 			}
 		} else if (strcmp(BASIL_ELM_RSVNARRAY, top) == 0) {
 			if (strcmp(BASIL_ELM_INVENTORY, prev) != 0) {
-				parse_err_illegal_start(d);
-				return (1);
-			}
-			if (brp->data.query.type != basil_query_inventory) {
-				parse_err_illegal_start(d);
-				return (1);
+				if (strcmp(BASIL_ELM_RESPONSEDATA, prev) != 0) {
+					parse_err_illegal_start(d);
+					return (1);
+				}
```

Comment:
Hi @prakashcv13 ,
    BASIL_ELM_RSVNARRAY can have two possible prev element. BASIL_ELM_INVENTORY or BASIL_ELM_RESPONSEDATA, that's why if it's not BASIL_ELM_INVENTORY we check for BASIL_ELM_RESPONSEDATA.

Below is a snippet from inventory query response:
</NodeArray>
   <ReservationArray/>
  </Inventory>
 </ResponseData>
</BasilResponse>